impl_core/src/main/java/io/opencensus/implcore/trace/propagation/B3Format.java

```diff
@@ -69,10 +95,21 @@
     checkNotNull(spanContext, "spanContext");
     checkNotNull(setter, "setter");
     checkNotNull(carrier, "carrier");
-    setter.put(carrier, X_B3_TRACE_ID, spanContext.getTraceId().toLowerBase16());
-    setter.put(carrier, X_B3_SPAN_ID, spanContext.getSpanId().toLowerBase16());
-    if (spanContext.getTraceOptions().isSampled()) {
-      setter.put(carrier, X_B3_SAMPLED, SAMPLED_VALUE);
+    if (getInjectionFormats().contains(B3InjectionFormat.SINGLE)) {
```

Comment:
An alternative approach would have been to put the injection logic in the enum. That would provide a compile-time guarantee that every case was implemented. This code would simplify to:

    for( final B3InjectionFormat format : getInjectionFormats() ) {
        format.inject( setter, carrier, spanContext );
    }

However, I opted not to go with that approach as `B3Injectionformat` is in the api module by necessity and it did not seem appropriate to put business logic there.